### PR TITLE
fix delete customized clusterClaim

### PR DESCRIPTION
**Pull request by zhiweiyin318**
issue: https://github.com/open-cluster-management/backlog/issues/13368

### CODE REVIEW

File: pkg/klusterlet/clusterclaim/clusterclaimer.go [hunk 1]
```diff
@@ -38,6 +38,7 @@ const (
 )
 
 const labelCustomizedOnly = "open-cluster-management.io/spoke-only"
+const labelCreatedBy = "open-cluster-management.io/created-by"
```
[the remark this answers is not in the document]
Comment (zhiweiyin318) [reply pkg/klusterlet/clusterclaim/clusterclaimer.go]: done, use` open-cluster-management.io/hub-managed`

File: test/e2e/clusterclaim_test.go [hunk 2]
```diff
@@ -1,108 +1,126 @@
...
-			return originalValue == value
-		}, eventuallyTimeout, eventuallyInterval).Should(gomega.BeTrue())
+		// should delete the claim synced from label
+		gomega.Eventually(func() error {
+			_, err := clusterClient.ClusterV1alpha1().ClusterClaims().Get(context.TODO(), "test", v1.GetOptions{})
+			if errors.IsNotFound(err) {
+				return nil
+			}
+			return fmt.Errorf("failed delete claim syned from label %v", err)
+		}, eventuallyTimeout, eventuallyInterval).ShouldNot(gomega.HaveOccurred())
```
Comment (qiujian16) [inline test/e2e/clusterclaim_test.go:124]: should we add an e2e to ensure user created claim will not be removed?

Comment (zhiweiyin318) [reply test/e2e/clusterclaim_test.go:124]: done

Comment (zhiweiyin318) [reply test/e2e/clusterclaim_test.go:124]: create a clusterClaim, test if get it in 1 min

**Pull request merged**